packaging_automation/common_tool_methods.py

```diff
@@ -290,3 +323,171 @@ def remove_cloned_code(exec_path: str):
         except:
             print(f"Some files could not be deleted in directory {exec_path}. "
                   f"Please delete them manually or they will be deleted before next execution")
+
+
+def process_template_file(project_version: str, templates_path: str, template_file_path: str):
+    '''Process template files and return the processed template '''
+    minor_version = get_minor_project_version(project_version)
+    env = get_template_environment(templates_path)
+    template = env.get_template(template_file_path)
+    return f"{template.render(project_version=project_version, project_minor_version=minor_version)}\n"
+
+
+def write_to_file(content: str, dest_file_name: str):
+    with open(dest_file_name, "w") as writer:
+        writer.write(content)
+
+
+def get_gpg_fingerprint_from_name(name: str):
+    '''Gets gpg key fingerprint from the key name. GPG fingerprint is unique parameter. However,
+    It is hard to track and reference in the code. When signing packages, gpg keys are referenced by name
+    and this function is used to get the unique identifier so fingerprint from name'''
+    result = subprocess.run(f"gpg --list-keys ", check=True, shell=True, stdout=subprocess.PIPE)
+    lines = result.stdout.decode("ascii").splitlines()
+    counter = 0
+    line_found = False
+    for line in lines:
+        if line.startswith("uid") and name in line:
+            line_found = True
+            break
+        counter = counter + 1
+    if not line_found:
+        return ""
+    else:
+        return lines[counter - 1].strip()
+
+
+def delete_public_gpg_key_by_name(name: str):
+    counter = 0
+
+    key_id = get_gpg_fingerprint_from_name(name)
+
+    if not key_id:
+        print(f"There is no key with name {name}. No need to delete ")
+        return
+
+    # There could be more than one key with the same name. While statement is used to delete all the public keys
+    # until no key remains (i.e. until exception arises)
+    while True:
+        output = run_with_output(f"gpg --batch --yes --delete-key {key_id}")
+        if output.returncode == 0:
+            print(f"Public key with the name {name} deleted")
+            counter = counter + 1
+        elif output.returncode == 2:  # key not exist
+            print(f"Key for the name {name} does not exist")
+            break
+        else:
+            print(f"Error {output.stderr.decode('ascii')}")
+            break
+        if counter > 0:
+            print(f"{counter} public  keys deleted")
```

Comment:
unified the implementation. output.returncode == 2 is important because key_id is the same for both private and public keys but sometimes one of them may be missing. To make sure to delete all the keys we need to continue on key non-existence case